---
title: "Notes: Coalescent Theory: John Wakeley"
author: "Piyush Agarwal"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
# url: your book url like https://bookdown.org/yihui/bookdown
# cover-image: path to the social sharing image like images/cover.jpg
description: This is a notes book on the Coalescent Theory by John Wakeley.
link-citations: yes
github-repo: https://github.com/piyushag0611/CoalescentTheoryNotes
---

# About

I am excited to note down my understanding from *Coalescent Theory: An Introduction* by *John Wakeley*. The book was recommended by my supervisor and was meant to be the first step in my thesis. Sadly, ignorantly I am still not there by the end of my first year. I did read stretches from the book, but did not make it cover to cover. I intend to wash away that mark and get a crucial first step to my thesis done. 

### Motivation {-}

There are a few fold objectives why I want to get some comprehensive set of notes written on the book:

1. It is to cement my own understanding. I know it is not enough to just read and have a claim of understanding. With no exercises, it is difficult to really make sure that I understand the content. I think that by **writing** down important pointers, I can make such a claim.

2. It will be a good written document for future purposes. I can advertise it for my own skills in the coalescent framework and refer to it in the future, given that there is no epdf of the book, and a library copy was difficult to retrieve.

3. Possibly helpful to my group. 

### Content {-}

The way I am joting down points are in the following way:

1. For metrics, distributions I am aiming to write down code, use sequences from NCBI or any other source and demonstrate the principals in the book. 

2. Mathematical expressions, equations will be derived from scratch. People not interested in derivations can simply skip them.

3. I will try to get a good sense of the purely theoretical ideas and make a decent case out of those ideas.

<!--chapter:end:index.Rmd-->

# Gene Genealogies

John Wakeley first talks about the coalescent framework and population genetics in general, and its development over the past century.

1.  Population Genetics aims to understand the **forces** that produce and maintain **genetic variation** among species:
    a.  Mutation
    b.  Recombination
    c.  Natural Selection
    d.  Population Structure
    e.  Random transmission of genetic material

<br>

2.  Coalescent Theory is a part of population genetics formulated to provide a mathematical framework for the analysis of *DNA sequence data*. So given a bunch of DNA sequences, the goal is to understand the population level processes that result in the variation observed in the sequences.

<br>

3.  Coalescent Theory deals with ancestry of the given sample of sequences and hence is a retrospective analysis (*backward in time*) of the variation than the classical prospective (*forward in time*) view of population genetics.

<br>

4.  The focus is on DNA sequences sampled from the **same genetic locus** and coming from a **single species** rather than a bunch of different species. The claim is that such a sample makes any study of the random effects of birth and death and other process within that population possible.

<br>

5.  (Remark from me) We assume that we are provided with a sample of well-assembled and aligned sequences, which is a very difficult task on its own.

## Genealogies and Genealogical Thinking

Genealogy is the set of *ancestral* relationship among the members of a sample.

<br>

For the first chapters we assume that sequences are extracted from regions without any recombination. With this assumption and another assumption that all life starts from a single common ancestor, the genealogy of any sample can be represented as a *rooted bifurcating tree*. Rooted because any sample has to share a common ancestor back in time (courtesy of the second assumption) and bifurcation because DNA replicates to give two copies. (*It is an interesting question in itself though, Are we really sure that DNA replicates into just exactly two copies or perhaps there was a time when it could have replicated into more copies?*)

```{r, echo=FALSE, fig.cap='Gene Genealogy with 5 samples: Bifurcated Tree with root not shown', fig.align='center', warning=FALSE}
source("plot_ms_trees_single_pop.R")
par(mar = c(4, 4, 0.1, 0.1))
plot_ms_tree(5, "-T -G 0.1")
```

5 Sequences {*s1, s2, s3, s4, s5*} are drawn at the current time $t=0$ and going back in time, pairs of lineages merge together. The internal nodes denote the common ancestors of the pair of lineages merging into it. Starting with $n$ samples, ($n=5$ in the above figure), there are $n-1$ internal nodes and a total of $2n-2$ branches with $n$ external branches *connecting the tips to the internal nodes* and $n-2$ internal branches connecting the internal nodes.

## Mutation and Mutation Model

1. Variation or polymorphism in genetic data reveals the genealogical structure among the members of the sample. If the sample consists of identical sequences, even though a genealogy exists, there is no way to identify or pin-point a genealogy. So mutations that bring variation in genetic data is said to be the bridge between genealogy and genetic data.

2. Mutations can be of various types:
    a. Single Nucleotide Polymorphism (SNPs): The sample differs in the nucleotide base at a particular site. Say a sample of 10 sequences might have 6 sequences with nucleotide 'A' on its 3rd site and the rest having a 'G' there.
    b. Insertions
    c. Deletions
    d. length variation
    
3. The book largely deals with SNPs with recombination being introduced only in the later chapters. 

4. Two mutation models are considered for the first few chapters:
    a. Infinite sites model
    b. Infinite alleles model

5. Infinite sites model - Every mutation (which we think of as a nucleotide substitution) happens at an independent site. So the number of sites with different bases in the sample is equal to the number of mutations that have taken place. 

6. Infinite alleles model - Every mutation introduces a new allele into the population. We can consider the infinite site model to subsume the infinite allele model with the consideration that sequences can only be differentiated and not precisely known. Every new haploid corresponds to a new allele.

7. One might question the appropriateness of the above two models, after all sequences are of finite length and mutations might happen repeatedly on a particular site. John Wakeley has tried to support it by pointing out that the variation is rather small, (polymorphic sites are in humans are of the order $10^{-3}$).

I think that the infinite site model maybe a good first iteration step, but one must consider their data. I work with viral sequences where the mutation rate is much higher and the infinite site model not so appropriate. But to understand the mutation process and how together with genealogy they affect variation in genetic data, the above models do help in that purpose.

## Quantifying Variation in Genetic Data

There are three quantities presented in the book to summarize and quantify genetic data:

1. No. of Segregating Sites in the sample $S$

2. Average pair-wise differences $\pi$

3. Site Frequency spectrum $\xi_i$

$S$ and $\pi$ are straight-forward and easy to understand. Note that $\pi$ can be written as:
$$\pi = \frac{1}{{n \choose 2}}\sum_{i=1}^n \sum_{j=i+1}^n k_{ij}$$

where $k_{ij}$ is the number of differences between sequence $i$ and sequence $j$. 
It is clear that these measures though simple and easily obtainable are rather drastic as they reduce all the data to single numbers. 
The Site Frequency spectrum $\xi_i$ gives a better view of the data. Suppose the ancestral sequence for the given sample is known. Then $\xi_i$ counts the number of sites with $i$ sequences having the mutant base and $n-i$ sequences having the ancestral base. Note that this only works when the infinite sites model is at play. (It will be ambiguous to assign any $\xi_i$ for any site undergoing multiple mutations). Also $\xi_i$ are known as the unfolded site-frequency spectrum, with the condition being that the ancestral sequence is known. With the ancestral sequence unknown, folded site-frequency spectrum ($\eta_i$) is  defined as: $$\eta_i = \frac{\xi_i + \xi_{n-i}}{1+\delta_{i, n-i}} \hspace{1cm} 1\leq i \leq  \left \lfloor \frac{n}{2} \right \rfloor$$

Also, $S$ and $\pi$ can be expressed in terms of $\xi_i$ as follows: 
$$\begin{align*} S =& \sum_{i=1}^{n-1}\xi_i \\ \pi =& \frac{1}{{n \choose 2}}\sum_{i=1}^{n-1}i(n-i)\xi_i \end{align*}$$
## Variation in a Real Dataset

The Human PDHA1 dataset is analyzed here. It is a sample consisting of sequences from the 4.2Kb region of the PDHA1 locus from a geographically diverse sample of 35 humans. 
















<!--chapter:end:01-chap.Rmd-->

# The Coalescent

After introducing and defining genealogies in the previous chapter, John Wakeley talks about the statistical properties of genealogies under the coalescent framework. To demonstrate the coalescent framework, he uses the classical population genetics models:

a.  Wright Fisher Model
b.  Moran Model

and shows these models converging to the Coalescent in the limit.

<br>

Throughout this chapter, there is no reference to variation in sequences with the following two reasons:

1.  A sample without any variation still supports a genealogy, because of the way DNA replication takes place

2.  All variation is assumed to be selectively neutral (any selection is dealt later in the book) and so there is no population advantage with respect to any particular variation.

## Basic Population Genetic Models

### Wright Fisher Model {.unnumbered}

-   Perfectly **non-overlapping** generations
-   Population size is constant and finite at all times
-   Each generation all current individuals die and are replaced by their offsprings. This is a random process with the possibility that a current individual has no offsprings in the next generation. For any individual in the next generation, any individual in the current generation has an equal chance of being a parent.

Let's consider a simple example of a population with two alleles **A** and **B** following the WF model. Let the population size be $N$ and $K_t$ be the no. of **A** alleles in generation $t$ with $K_0 = i$. Then the probability that the $K_1 = j$ is given by: $$\mathbb{P}_{ij} = {N \choose j}\left(\frac{i}{N}\right)^j\left(1-\frac{i}{N}\right)^{N-j}$$ We also have that the expected number and variance of **A** alleles in generation $1$ as $$\begin{align*} & \mathbb{E}(K_1) = N\left(\frac{i}{N}\right) = i \\ & \mathbb{V}ar(K_1) = N \left(\frac{i}{N}\right) \left(1 - \frac{i}{N}\right) = i\left(1-\frac{i}{N}\right)  \end{align*}$$

The expected number of **A** alleles remains unchanged over time.

### Moran Model {.unnumbered}

-   Perfectly **Overlapping** Generations
-   Population size is again constant and finite over time
-   At the end of every generation, two current individuals are chosen. One individual survives and gives rise to an offspring where as the other perishes.

Consider the same example with the population following the Moran Model instead of the WF Model. Again let $K_t$ be the no. of **A** allele in generation $t$. Then with $K_0=i$, probability that $K_1=j$ is given by: $$ \mathbb{P}_{ij} = \begin{cases} p(1-p) & j=i+1 \\ (1-p)p & j=i-1 \\ p^2 + (1-p)^2 & j=i \\ 0 & \text{otherwise} \end{cases}    $$

where ($p=\frac{i}{N}$). Again the expected number and variance of no. of **A** alleles is given by: $$\begin{align*}  \mathbb{E}(K_1) =& p(1-p)(i+1) + p(1-p)(i-1) + (p^2+(1-p)^2)i \\ =& i \\  \mathbb{V}ar(K_1) =& p(1-p)(i+1)^2 + p(1-p)(i-1)^2 + (p^2+(1-p)^2)i^2  -i^2 \\ =& 2p(1-p)  \end{align*}$$

There are two steps to cross-reference any heading:

1.  Label the heading: `# Hello world {#nice-label}`.
    -   Leave the label off if you like the automated heading generated based on your heading title: for example, `# Hello world` = `# Hello world {#hello-world}`.
    -   To label an un-numbered heading, use: `# Hello world {-#nice-label}` or `{# Hello world .unnumbered}`.
2.  Next, reference the labeled heading anywhere in the text using `\@ref(nice-label)`; for example, please see Chapter.
    -   If you prefer text as the link instead of a numbered reference use: [any text you want can go here].

## Captioned figures and tables

Figures and tables *with captions* can also be cross-referenced from elsewhere in your book using `\@ref(fig:chunk-label)` and `\@ref(tab:chunk-label)`, respectively.

See Figure \@ref(fig:nice-fig).

```{r nice-fig, fig.cap='Here is a nice figure!', out.width='80%', fig.asp=.75, fig.align='center', fig.alt='Plot with connected points showing that vapor pressure of mercury increases exponentially as temperature increases.'}
par(mar = c(4, 4, .1, .1))
plot(pressure, type = 'b', pch = 19)
```

Don't miss Table \@ref(tab:nice-tab).

```{r nice-tab, tidy=FALSE}
knitr::kable(
  head(pressure, 10), caption = 'Here is a nice table!',
  booktabs = TRUE
)
```

<!--chapter:end:02-chap.Rmd-->

# Neutral Genetic Variation

You can add parts to organize one or more book chapters together. Parts can be inserted at the top of an .Rmd file, before the first-level chapter heading in that same file. 

Add a numbered part: `# (PART) Act one {-}` (followed by `# A chapter`)

Add an unnumbered part: `# (PART\*) Act one {-}` (followed by `# A chapter`)

Add an appendix as a special kind of un-numbered part: `# (APPENDIX) Other stuff {-}` (followed by `# A chapter`). Chapters in an appendix are prepended with letters instead of numbers.




<!--chapter:end:03-chap.Rmd-->

# The Structured Coalescent

## Footnotes

Footnotes are put inside the square brackets after a caret `^[]`. Like this one ^[This is a footnote.]. 

## Citations

Reference items in your bibliography file(s) using `@key`.

For example, we are using the **bookdown** package [@R-bookdown] (check out the last code chunk in index.Rmd to see how this citation key was added) in this sample book, which was built on top of R Markdown and **knitr** [@xie2015] (this citation was added manually in an external file book.bib). 
Note that the `.bib` files need to be listed in the index.Rmd with the YAML `bibliography` key.


The RStudio Visual Markdown Editor can also make it easier to insert citations: <https://rstudio.github.io/visual-markdown-editing/#/citations>

<!--chapter:end:04-chap.Rmd-->

# Separation of Time Scales

## Equations

Here is an equation.

\begin{equation} 
  f\left(k\right) = \binom{n}{k} p^k\left(1-p\right)^{n-k}
  (\#eq:binom)
\end{equation} 

You may refer to using `\@ref(eq:binom)`, like see Equation \@ref(eq:binom).


## Theorems and proofs

Labeled theorems can be referenced in text using `\@ref(thm:tri)`, for example, check out this smart theorem \@ref(thm:tri).

::: {.theorem #tri}
For a right triangle, if $c$ denotes the *length* of the hypotenuse
and $a$ and $b$ denote the lengths of the **other** two sides, we have
$$a^2 + b^2 = c^2$$
:::

Read more here <https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html>.

## Callout blocks


The R Markdown Cookbook provides more help on how to use custom blocks to design your own callouts: https://bookdown.org/yihui/rmarkdown-cookbook/custom-blocks.html

<!--chapter:end:05-chap.Rmd-->

# Ancestral Graphs

## Publishing

HTML books can be published online, see: https://bookdown.org/yihui/bookdown/publishing.html

## 404 pages

By default, users will be directed to a 404 page if they try to access a webpage that cannot be found. If you'd like to customize your 404 page instead of using the default, you may add either a `_404.Rmd` or `_404.md` file to your project root and use code and/or Markdown syntax.

## Metadata for sharing

Bookdown HTML books will provide HTML metadata for social sharing on platforms like Twitter, Facebook, and LinkedIn, using information you provide in the `index.Rmd` YAML. To setup, set the `url` for your book and the path to your `cover-image` file. Your book's `title` and `description` are also used.



This `gitbook` uses the same social sharing data across all chapters in your book- all links shared will look the same.

Specify your book's source repository on GitHub using the `edit` key under the configuration options in the `_output.yml` file, which allows users to suggest an edit by linking to a chapter's source file. 

Read more about the features of this output format here:

https://pkgs.rstudio.com/bookdown/reference/gitbook.html

Or use:

```{r eval=FALSE}
?bookdown::gitbook
```



<!--chapter:end:06-chap.Rmd-->

# Simulations and Inference

Notes to be added

<!--chapter:end:07-chap.Rmd-->

